mojom: Drop using imp module.

This module is gone in python 3.12 onwards, in most places it is unused
so remove from those places, in some places where its still is needed
replace it with importlib.util

Fixes build issues with current version of mojom vendored into chromium used inside qtwebengine
this wont be needed once the mojom is upgraded to latest

Signed-off-by: Khem Raj <raj.khem at gmail.com>
---
 chromium/mojo/public/tools/mojom/mojom/fileutil.py        | 1 -
 .../ipc/mojo/public/tools/mojom/mojom/fileutil_unittest.py | 1 -
 .../tools/mojom/mojom/generate/generator_unittest.py       | 7 ++-----
 .../tools/mojom/mojom/generate/translate_unittest.py       | 1 -
 .../mojo/public/tools/mojom/mojom/parse/ast_unittest.py    | 1 -
 .../mojom/mojom/parse/conditional_features_unittest.py     | 7 ++-----
 chromium/mojo/public/tools/mojom/mojom/parse/lexer.py     | 1 -
 .../mojo/public/tools/mojom/mojom/parse/lexer_unittest.py  | 6 ++----
 .../mojo/public/tools/mojom/mojom/parse/parser_unittest.py | 1 -
 9 files changed, 6 insertions(+), 20 deletions(-)

--- a/chromium/mojo/public/tools/mojom/mojom/fileutil_unittest.py
+++ b/chromium/mojo/public/tools/mojom/mojom/fileutil_unittest.py
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import imp
 import os.path
 import shutil
 import sys
--- a/chromium/mojo/public/tools/mojom/mojom/generate/generator_unittest.py
+++ b/chromium/mojo/public/tools/mojom/mojom/generate/generator_unittest.py
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import imp
+import importlib.util
 import os.path
 import sys
 import unittest
@@ -18,10 +18,7 @@ def _GetDirAbove(dirname):
     if tail == dirname:
       return path
 
-
-try:
-  imp.find_module("mojom")
-except ImportError:
+if importlib.util.find_spec("mojom") is None:
   sys.path.append(os.path.join(_GetDirAbove("pylib"), "pylib"))
 from mojom.generate import generator
 
--- a/chromium/mojo/public/tools/mojom/mojom/generate/translate_unittest.py
+++ b/chromium/mojo/public/tools/mojom/mojom/generate/translate_unittest.py
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import imp
 import os.path
 import sys
 import unittest
--- a/chromium/mojo/public/tools/mojom/mojom/parse/ast_unittest.py
+++ b/chromium/mojo/public/tools/mojom/mojom/parse/ast_unittest.py
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import imp
 import os.path
 import sys
 import unittest
--- a/chromium/mojo/public/tools/mojom/mojom/parse/conditional_features_unittest.py
+++ b/chromium/mojo/public/tools/mojom/mojom/parse/conditional_features_unittest.py
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import imp
+import importlib.util
 import os
 import sys
 import unittest
@@ -18,10 +18,7 @@ def _GetDirAbove(dirname):
     if tail == dirname:
       return path
 
-
-try:
-  imp.find_module('mojom')
-except ImportError:
+if importlib.util.find_spec("mojom") is None:
   sys.path.append(os.path.join(_GetDirAbove('pylib'), 'pylib'))
 import mojom.parse.ast as ast
 import mojom.parse.conditional_features as conditional_features
--- a/chromium/mojo/public/tools/mojom/mojom/parse/lexer_unittest.py
+++ b/chromium/mojo/public/tools/mojom/mojom/parse/lexer_unittest.py
@@ -2,7 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import imp
+import importlib.util
 import os.path
 import sys
 import unittest
@@ -22,9 +22,7 @@ def _GetDirAbove(dirname):
 sys.path.insert(1, os.path.join(_GetDirAbove("mojo"), "third_party"))
 from ply import lex
 
-try:
-  imp.find_module("mojom")
-except ImportError:
+if importlib.util.find_spec("mojom") is None:
   sys.path.append(os.path.join(_GetDirAbove("pylib"), "pylib"))
 import mojom.parse.lexer
 
--- a/chromium/mojo/public/tools/mojom/mojom/parse/parser_unittest.py
+++ b/chromium/mojo/public/tools/mojom/mojom/parse/parser_unittest.py
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import imp
 import os.path
 import sys
 import unittest
